### PR TITLE
Merge with branch gabriel

### PR DESCRIPTION
**Pull request by gdumouli**
NoRAMApp added to share data across activities.
UI of event-pages for attendee finished.
ArrayAdapter for events created.

### CODE REVIEW

File: code/noRAM/app/src/main/java/com/example/noram/AttendeeEventInfo.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+import com.example.noram.model.Event;
+
+public class AttendeeEventInfo extends AppCompatActivity {
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/AttendeeEventInfo.java:18]: add java docs

File: code/noRAM/app/src/main/java/com/example/noram/AttendeeEventListFragment.java [hunk 2]
```diff
@@ -51,10 +80,184 @@ public void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    /**
+     * Makes the user's personal events visible and hides the other lists
+     */
+    public void displayMyEvents(){
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/AttendeeEventListFragment.java:86]: docs

File: code/noRAM/app/src/main/java/com/example/noram/AttendeeEventListFragment.java [hunk 3]
```diff
@@ -23,6 +41,17 @@ public class AttendeeEventListFragment extends Fragment {
...
+    private ListView allEventList; // list of all events in UI
+    private ListView userEventList; // list of all user's events in UI
+    private ListView searchEventList; // list of events' search results
+    private ArrayList<Event> allEventDataList; // data list of all events
+    private ArrayList<Event> userEventDataList; // data list of all user's events
+    private ArrayList<Event> searchEventDataList; // data list of events' search results
+    private EditText searchInput; // searchbar
+    private boolean viewingUserEvents; // indicates if showing all events or just user' events
 
     public AttendeeEventListFragment() {
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/AttendeeEventListFragment.java:58]: docs

File: code/noRAM/app/src/main/java/com/example/noram/controller/EventArrayAdapter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.example.noram.R;
+import com.example.noram.model.Event;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+public class EventArrayAdapter extends ArrayAdapter<Event> {
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/controller/EventArrayAdapter.java:22]: docs

File: code/noRAM/app/src/main/java/com/example/noram/model/Event.java
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/model/Event.java:1]: add an updateDB method that is called inside each setter please

File: code/noRAM/app/src/main/java/com/example/noram/model/Event.java [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+     */
+    public void setPromoQR(Photo promoQR) {
+        this.promoQR = promoQR;
+    }
+
+    /**
+     * Set whether or not check-in location is tracked
+     * @param trackLocation new trackLocation for event
+     */
+     public void changeTrackLocation(boolean trackLocation) {
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/model/Event.java]: should be setTrackLocation just so its standard

File: code/noRAM/app/src/main/java/com/example/noram/model/NoRAMApp.java
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/model/NoRAMApp.java:1]: remove how you are currently working with the db as we have changed it to be a static reference accessible across all files. please refer to current master or chat with me

**Review by WhoShotCupidd — CHANGES_REQUESTED** [review]
please check my comments on whats needed

File: code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+public class AttendeeAnnouncementsActivity extends AppCompatActivity {
+    private static final String eventIDLabel = "eventID";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState){
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.attendee_announcements);
+
+        // retrieve corresponding event in database
+        int eventID = getIntent().getIntExtra(eventIDLabel,0);
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java:18]: Whats this getIntent we use storage refs for the db

**Review by WhoShotCupidd — CHANGES_REQUESTED** [review]
check the getIntent comment please if that can be fixed its all good to merge

File: code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java (hunk 6 above)
Comment (gdumouli) [reply code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java:18]: The getIntent is to get the event whose notifications we want to display. 

AttendeeAnnouncementsActivity gets called when someone clicks on an Event's notifications, where the Event activity will swap with Announcements activity while specifying which event notifications we want to display with the Intent

Comment (WhoShotCupidd) [reply code/noRAM/app/src/main/java/com/example/noram/AttendeeAnnouncementsActivity.java:18]: ahhhh i see

**Review by WhoShotCupidd — APPROVED** [review]
lgtm

**Pull request merged**